src/app/core/models/asset/asset-property.model.ts

```diff
@@ -29,19 +36,19 @@ export abstract class AssetProperty {
     return this.assetId;
   }
 
-  isAddress(): boolean {
+  isAddress(): this is Address {
     return this.getType() === AssetPropertyType.Address;
   }
 
-  isDefendant(): boolean {
+  isDefendant(): this is Defendant {
     return this.getType() === AssetPropertyType.Defendant;
   }
 
-  isSolution(): boolean {
+  isSolution(): this is Solution {
     return this.getType() === AssetPropertyType.Solution;
   }
 
-  isStorageSpace(): boolean {
+  isStorageSpace(): this is StorageSpace {
```

Comment:
What is this sorcery? Please change it back to `boolean` so we have proper type hinting.